core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -109,4 +158,50 @@ public int hashCode() {
         result = 31 * result + Long.hashCode(generation);
         return result;
     }
+
+    /** A copy of {@link OutputStreamIndexOutput} without the intermediate buffering */
+    static class DirectOutputStreamIndexOutput extends IndexOutput {
```

Comment:
Because it  bugged me that we do buffering when we write to an in-memory buffer anyway. I'm not sure why that class does buffering when you can give it a buffering stream, if you want buffering? Alternatively we should allow to set the buffer size to 0 , in which cases the CRC will not be wrapped.